quinn-proto/src/connection/paths.rs

```diff
@@ -37,6 +40,11 @@ pub(super) struct PathData {
     /// Used in persistent congestion determination.
     pub(super) first_packet_after_rtt_sample: Option<(SpaceId, u64)>,
     pub(super) in_flight: InFlight,
+    /// Whether this path has had it's remote address reported back to the peer. This only happens
+    /// if both peers agree to so based on their transport parameters.
+    pub(super) observed_addr_sent: bool,
```

Comment:
Could we avoid tracking this extra state by queueing a frame immediately whenever we change the path?